opencopilot/utils/langchain/llm_GPT.py

```
@@ -28,13 +49,17 @@ def extract_json_block(text):
 
 
 def run(messages, llm_names):
+    global llm
     model = None
 
+    # Select the first found configured model
     for llm_model in llm_names:
         if llm_model in llm_configs:
             model = llm_model
             break
 
+    engine = GPT3_ENGINE if "GPT3" in model else GPT4_ENGINE
+
     if model is None:
         print("Didn't find configurations of any of the desired models, Please set the configuration of the desired "
```

Define a class for that exception in exceptions.py, raise it here instead of the print.